src/filter/mod.rs

```diff
@@ -26,17 +26,21 @@ pub enum FilterError {
     /// Specific to a ProgrammableFilter, means no function is available as
     /// called in the script
     NoSuchFunction(&'static str, metric::Event),
+    /// Specific to a ProgrammableFilter, means the lua code errored
+    LuaError(String, metric::Event),
 }
 
-fn name_in_fe(fe: &FilterError) -> &'static str {
-    match *fe {
-        FilterError::NoSuchFunction(n, _) => n,
+fn msg_in_fe(fe: &FilterError) -> &str {
+    match fe {
```

Comment:
This could be `match *fe` and then you wouldn't need to match on references.

Reply:
That's how it used to work, but I hit issues with lifetime because LuaError contains a `String`, not a `&'static str`. Trying to return a borrow of the deconstructed `String` said that it didn't live past the end of this function.

Reply:
Oh shoot, of course.